feat(badges): first batch of design updates for the badge component as part of SHINE

mukunku opened the pull request:
# Summary

This is for [SPARK-59](https://stackoverflow.atlassian.net/browse/SPARK-59) and updates the Badge component styles based on the [new designs](https://www.figma.com/design/do4Ug0Yws8xCfRjHe9cJfZ/Project-SHINE---Product-UI?node-id=610-18799&p=f&m=dev). 

<img width="150" alt="image" src="https://github.com/user-attachments/assets/47e8f63b-0619-47b1-b065-553a4107a801" />

This PR should only be merged after #2013 is merged as it was branched off of that branch.

# Changes

The PR does the following:
* Updates styles for **Default** badges
* Removes the **Badge counts** section
* Adds a new **Tag badges** section
* Updates **User badge** designs
* Removes `xs` size and replaces it with `lg`

What is **NOT** in scope for this PR:
* **Number count** badges
* **Icon badges** 
* **Badge states**
* **Filled badges**
* Cleaning up the Svelte API for the Badge component

# Breaking Changes

* `xs` size was removed
* `s-award-bling` class was renamed to `s-bling` (used in some badge templates)

# How to Test

Checkout the new styles in Stacks Docs and Stacks Svelte:

* Stacks Docs: https://deploy-preview-2051--stacks.netlify.app/product/components/badges/
* Stacks Svelte: https://deploy-preview-2051--stacks-svelte.netlify.app/?path=/docs/components-badge--docs

# How to Review

I recommend excluding all `.ico` files in the GitHub file diff for a nice reviewing experience:
<img width="150"  src="https://github.com/user-attachments/assets/76c50f29-09bf-4bc6-b36d-764d2918d57e" />

dancormier committed:
chore(bling): update to SHINE styling

dancormier committed:
Let's stick with "award bling" for now

dancormier committed:
Why was I doing `--`?

dancormier committed:
update tests

dancormier committed:
Update other baseline images

dancormier committed:
Update svelte component

dancormier committed:
update tests, formatting

dancormier committed:
Fix some tests

dancormier committed:
got a little lint in my pocket

dancormier committed:
Change name to bling

dancormier committed:
formatting!

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Add TODO for migration guide

dancormier committed:
Address PR review

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Remove award bling test images

dancormier committed:
update remaining visual regression test images

dancormier committed:
Update activity variant to use pink colors

dancormier committed:
changeset

dancormier committed:
Address review

dancormier committed:
Update docs, tests to use updated badge, migration guide

dancormier committed:
whoops

dancormier committed:
Add svelte link

dancormier committed:
Change rep bling color

dancormier committed:
Omit rep, activity base bling from svelte examples

dancormier committed:
Make the base bling appear as the base bling in svelte docs

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Update docs

mukunku committed:
basic implementation of new styles

mukunku committed:
update svelte component for new badge styles

mukunku committed:
update baseline

mukunku committed:
lint

mukunku committed:
moar linting

mukunku committed:
moar linting

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

mukunku committed:
update baseline

mukunku committed:
add theming back

mukunku committed:
make only admin badge themeable

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

mukunku commented on packages/stacks-classic/lib/components/badge/badge.less:
```diff
@@ -117,48 +117,36 @@
     }
 
     &&__moderator {
-        --_ba-bc: var(--theme-secondary-300);
```
The mod badges aren't customizable today in SOE it seems: 
<img width="200" alt="image" src="https://github.com/user-attachments/assets/f59fbbb6-6da9-4dbb-9214-026a2245cb9a" />

So we should be okay to not make the new mod badge themeable either. [Thread here](https://stackexchange.slack.com/archives/C096063B1PH/p1763402030395109?thread_ts=1763400519.055879&cid=C096063B1PH)

mukunku committed:
fix copy

mukunku committed:
changeset+migration_guide

mukunku committed:
fix high contrast

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

mukunku committed:
update baseline

mukunku committed:
Merge branch 'sal/SPARK-59-2' of https://github.com/StackExchange/Stacks into sal/SPARK-59-2

CGuindon reviewed (changes requested):
Looking really good overall, the only issue I'm seeing is with the sizes and it's related to how we did the sizing on the `clear-bling` component so I'm looping in @dancormier as we might need a separate PR to tweak those a bit (my fault for not catching that initially).

Using the tag badge as an example, it's using `s-bling s-bling__gold` which is the bling shape without a background. 

1. When I set the badge to use `s-badge__sm`, the bling within it should also be using  the bling to using `s-bling__sm` — not sure if this can be automated or if we just need to make sure that core devs apply the `sm` variant to both components. **We should add an example in the docs for this — expand on the Sizes section and show examples for the different variants**

---

2. The overall size of a tag badge with `s-badge__sm` and the padding around the bling inside is too big because it's getting thrown off by the size of the `s-bling__sm` (also too big).
    * The overall size of `s-bling__sm s-bling__filled` is 16x16px (this is correct)
    * The overall size of  `s-bling__sm s-bling__gold` should be 8x8px and the shape inside that bounding box is 7x7px (gold and silver are 7x7 while bronze square and the dot are 6x6)
    * I think if `s-bling__sm s-bling__gold` gets it's proper 8x8px size then `s-badge__sm` should also shrink the overall height and fix the padding surrounding the bling to be smaller.
    
<img width="360" height="164" alt="Screenshot 2025-11-18 at 8 17 27 AM" src="https://github.com/user-attachments/assets/e0b521dd-fcb1-4ced-996f-eb73db20616f" />

---

3. Same issue for the large size of clear (default) bling. The little shapes are keeping the larger bounding box that `filled` bling have and aren't growing in shape.

We can make this all a separate ticket, fix bling and then in that PR make sure that the badge sizes work too.

mukunku committed:
fix bling sizing issues

mukunku committed:
add more size options to docs

mukunku committed:
replace all s-bling__xs usages with s-bling__sm

mukunku committed:
fix svelte component

mukunku committed:
lint

mukunku commented:
Thanks for the discussion @CGuindon & @dancormier! I tweaked the Bling component as we discussed based on the designs:

* Bling component unfilled small size will be 8x8px. With the icon size 6x6px, except silver and gold which will be 7x7px
* Bling component unfilled large and regular sizes will be 10x10px. With the icon size as 8x8px, except silver and gold which will be 9x9px.

I also added more size examples to Stacks Docs now: https://deploy-preview-2051--stacks.netlify.app/product/components/badges/#badge-sizes

Note: I find testing the different variants easier from the [Svelte component side](https://deploy-preview-2051--stacks-svelte.netlify.app/?path=/docs/components-badge--docs).

CGuindon commented:
(not fully related) I noticed if you inspect the example of `s-bling` ([link](https://deploy-preview-2051--stacks.netlify.app/product/components/bling/#types)) it has some extra broken code class in there that's not doing anything. We can delete the `s-bling__`
<img width="439" height="176" alt="Screenshot 2025-11-18 at 1 59 52 PM" src="https://github.com/user-attachments/assets/2e4349ab-b569-4a06-9848-e3aed4a7b866" />
<img width="846" height="853" alt="Screenshot 2025-11-18 at 2 01 35 PM" src="https://github.com/user-attachments/assets/2cbd32e5-2a57-4ff6-9d88-effbd8302c45" />

CGuindon reviewed (approved):
I think this all works! Any issues I'm seeing are not related to this PR/work (issues with colors). so I'm going to approve this and move on to the next!

giamir commented:
@mukunku @dancormier Should we expose the Svelte component to consumers as part of this update?
<img width="1038" height="290" alt="Screenshot 2025-11-19 at 09 30 43" src="https://github.com/user-attachments/assets/ba31af98-99d7-4a0f-8a7a-cc1d503b46a3" />

If we are showing it in storybook we probably should exposing it in the `index.ts` as well I think.

dancormier committed:
Merge branch 'beta' into sal/SPARK-59-2

dancormier committed:
Minor bling less, docs cleanup

dancormier committed:
Styling, docs updates

dancormier reviewed (changes requested):
Thanks for taking this on @mukunku and thanks for the bling fixes ❤️ Since many variants are yet to be designed (I saw a banner in the [Figma](https://www.figma.com/design/do4Ug0Yws8xCfRjHe9cJfZ/Project-SHINE---Product-UI?node-id=1397-33319&m=dev) saying "Variants below will be done separately — need additional mock ups first"), I took this opportunity to remove them from the Less for now and do a refactor (see https://github.com/StackExchange/Stacks/pull/2051/commits/4132cad26eca45c15f9d31908e74d20d61dc50cb). 

I agree w/ @giamir that we should expose the Svelte component as well to consumers. It'll need a some updates (most notably, I think "award" can be merged into "variant") but I think the changes won't be _too_ crazy. Let me know if you'd like help with that or if you want any opinions on the API.

dancormier commented on packages/stacks-classic/lib/components/badge/badge.less:
```diff
@@ -1,269 +1,114 @@
+    --_ba-fw: unset;
+    --_ba-g: var(--su6);
+    --_ba-mnh: var(--su24);
+    --_ba-px: var(--su6); // TODO SHINE verify w/ design this is ok for default, gold, rep, activity (Figma show it w/ 8px right padding for only those variants)
```
@CGuindon I noticed a couple of variants of the default sized badge with filled bling (gold, default, rep, activity) had right padding of 8px, while the rest had 6px.

Should we be using 6px for all of them?


### padding-right: 8px

<img width="410" height="250" alt="Image" src="https://github.com/user-attachments/assets/bf063b5b-5ead-437e-862a-f93cfbe51a57" />

### padding-right: 6px

<img width="390" height="176" alt="Image" src="https://github.com/user-attachments/assets/3f7a90c4-1423-437e-8e09-4107809f009b" />

mukunku replied:
Discussed, they should all be consistent across sizes

dancormier committed:
Update more visual regression tests

mukunku committed:
Revert "Update more visual regression tests"

This reverts commit cb5344b3aa198b1b4232c146be1fd6fb0f84550b.

mukunku committed:
Revert "Styling, docs updates"

This reverts commit 4132cad26eca45c15f9d31908e74d20d61dc50cb.

mukunku committed:
Revert "Minor bling less, docs cleanup"

This reverts commit d5108cef38a632bcf692cb250ed8d574ce9f69b1.

mukunku committed:
pr feedback

mukunku committed:
expose svelte badge component

mukunku committed:
lint

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

mukunku committed:
update line-height

mukunku commented:
Hey @dancormier , can you kindly take a look at the PR again when you get the chance?

As discussed I made the following changes:
* Reverted your commits 🙃 (Happy to help with those changes as a follow-up 🙏🏼 )
* Fixed font-weight issue for rep badges (thanks for catching).
* Switched line-height from `126%` to `var(--lh-md)` which is `128%` it seems.
* Adjusted Badge Size examples section:
   <img width="300" height="198" alt="{219E5F1B-EDE5-488C-9207-D8F0CA7FDB38}" src="https://github.com/user-attachments/assets/e70b7e98-2cb2-499e-9b8e-b4b78d23ec96" /> 

I also exposed the Badge Svelte component in [index.ts](https://github.com/StackExchange/Stacks/pull/2051/commits/0040cc4e87d46df2248dee0eed22c2c5372592ff) as mentioned by Giamir.

mukunku committed:
update baseline

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

mukunku committed:
fix lint

mukunku committed:
remove comments

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2

dancormier commented on packages/stacks-docs/product/components/badges.html:
Since the Svelte component is now exposed to consumers, we should add the svelte link and update the [figma link ](https://www.figma.com/design/do4Ug0Yws8xCfRjHe9cJfZ/Project-SHINE---Product-UI?node-id=610-18799&p=f&t=u4su1MPgebbjmcfI-0) here

mukunku replied:
Thanks, Dan. I forgot about those: https://github.com/StackExchange/Stacks/pull/2051/commits/3a6647dda761fd5c5b106e1ce8161890f4dc528b

I guess we'll update all the svelte links after v3 is live as they all point to the beta url at the moment?

dancormier reviewed (approved):
It's looking good @mukunku! I think this is in a good place to be merged. I have one tiny suggestion but otherwise it's good to go. Thanks for the great work and your patience ❤️

mukunku committed:
add svelte badge and update figma link

mukunku committed:
Merge branch 'beta' into sal/SPARK-59-2
